packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -11,14 +16,106 @@ const Separator = styled(Toolbar.Separator)`
   height: ${pxToRem(24)};
 `;
 
+/**
+ *
+ * @param {import('slate').BaseEditor} editor
+ * @param {string} name - name of the modifier
+ */
+const isModifierActive = (editor, name) => {
+  const modifiers = Editor.marks(editor);
+
+  return modifiers ? modifiers[name] === true : false;
+};
+
+/**
+ *
+ * @param {import('slate').BaseEditor} editor
+ * @param {string} name - name of the modifier
+ */
+const toggleModifier = (editor, name) => {
+  const isActive = isModifierActive(editor, name);
+
+  if (isActive) {
+    Editor.removeMark(editor, name);
+  } else {
+    Editor.addMark(editor, name, true);
+  }
+};
```

Comment:
How come these are these defined outside the component scope?

Reply:
I was wondering the same 🤔

Reply:
tbh I mostly followed one of the official slate examples: https://github.com/ianstormtaylor/slate/blob/main/site/examples/richtext.tsx#L111-L126

Would there be a benefit in moving it inside the component?

Reply:
I moved them inside the component, now I don't have to pass the editor everywhere